topics/client.md

```diff
@@ -1,105 +1,204 @@
+<var name="artifact_name" value="ktor-client-core"/>
+<include src="lib.md" include-id="add_ktor_artifact"/>
 
-The first thing you need to do before using the client is to add a client engine dependency. Client engine is a request executor that performs requests from ktor API. There are many client engines for each platform available out of the box: [`Apache`](http-client_engines.md#apache),
-[`OkHttp`](http-client_engines.md#okhttp),
-[`Android`](http-client_engines.md#android),
-[`Ios`](http-client_engines.md#ios),
-[`Js`](http-client_engines.md#js-javascript),
-[`Jetty`](http-client_engines.md#jetty),
-[`CIO`](http-client_engines.md#cio) and [`Mock`](http-client_engines.md#mock). For a full list of dependencies required for a specific engine, see [](http-client_engines.md#dependencies). 
 
-For example, you can add a `CIO` engine dependency as follows:
+### Engine Dependency {id="engine-dependency"}
+An engine is in charge of processing network requests. There are different client engines available for [various platforms](http-client_multiplatform.md), such as Apache, CIO, Android, iOS, and so on. For example, you can add a `CIO` engine dependency as follows:
 <var name="artifact_name" value="ktor-client-cio"/>
 <include src="lib.md" include-id="add_ktor_artifact"/>
 
-## Creating client
+For a full list of dependencies required for a specific engine, see [](http-client_engines.md#dependencies).
+
+### Feature Dependency {id="feature-dependency"}
+If you want to use additional client features, you need to add a corresponding dependency. You can learn which dependencies you need from a topic for a required feature.
+
 
-Next you can create a client as here:
+
+
+## Create the Client {id="create-client"}
+
+To instantiate the client, create the [HttpClient](https://api.ktor.io/%ktor_version%/io.ktor.client/-http-client/index.html) class instance and pass an engine as a parameter:
 
 ```kotlin
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+
 val client = HttpClient(CIO)
 ```
 
-where `CIO` is engine class here. If you confused which engine class you should use consider using `CIO`.
-
-If you're using multiplatform, you can omit the engine:
+In this example, we use the [CIO](https://api.ktor.io/%ktor_version%/io.ktor.client.engine.cio/-c-i-o/index.html) engine. You can also omit an engine:
 
 ```kotlin
+import io.ktor.client.*
+
 val client = HttpClient()
 ```
 
-Ktor will choose an engine among the ones that are available from the included artifacts using a `ServiceLoader` on the JVM, or similar approach in the other platforms. If there are multiple engines in the dependencies Ktor chooses first in alphabetical order of engine name.
+In this case, the client will choose an engine automatically depending on the artifacts [added in a build script](#engine-dependency). You can learn more from the [Engines](http-client_engines.md#default) topic.
```

Comment:
"You can learn more from the Engines topic" doesn't sound quite right.

Reply:
what's wrong here? I need to clarify what a reader can learn?

Reply:
I find it the construct somewhat strange. What are they meant to be learning more on? What engines are or how they work? 

If it's about what engines are, then I'd suggest something "If you want to learn more about engines, please check the [Engine topic]

although I'd replace topic for documentation TBH.

Reply:
ok, I see. Changed to 'You can learn how the client chooses an engine from the *** documentation section.'